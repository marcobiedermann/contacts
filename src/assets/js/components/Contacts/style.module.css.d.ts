declare const styles: {
  readonly contacts__list: string;
  readonly contacts__item: string;
};
export = styles;
