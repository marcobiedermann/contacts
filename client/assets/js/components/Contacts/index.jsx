import PropTypes from 'prop-types';
import React from 'react';
import ContactPreview from '../ContactPreview';
import styles from './style.css';

const Contacts = props => {
  const { contacts } = props;

  return (
    <div className={styles.contacts}>
      <ul className={styles.contacts__list}>
        {contacts.map(contact => (
          <li className={styles.contacts__item} key={contact.id}>
            <ContactPreview {...contact} />
          </li>
        ))}
      </ul>
    </div>
  );
};

Contacts.propTypes = {
  contacts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
    }),
  ),
};

Contacts.defaultProps = {
  contacts: [],
};

export default Contacts;
