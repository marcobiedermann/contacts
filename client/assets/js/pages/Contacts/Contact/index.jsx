import PropTypes from 'prop-types';
import React from 'react';
import { Route, Switch } from 'react-router-dom';
import EditPage from './Edit';
import ContactDetail from '../../../components/ContactDetail';

const ContactPage = ({
  contacts,
  match,
  updateContact,
  removeContact,
}) => {
  const contact = contacts.find(c => c.id === match.params.id);

  return (
    <Switch>
      <Route
        path="/contacts/:id/edit"
        component={() => (
          <EditPage
            contact={contact}
            updateContact={updateContact}
          />
        )}
      />
      <Route
        path="/contacts/:id"
        component={() => (
          <ContactDetail
            {...contact}
            removeContact={removeContact}
          />
        )}
      />
    </Switch>
  );
};

ContactPage.propTypes = {
  contacts: PropTypes.arrayOf(PropTypes.shape()),
  match: PropTypes.shape(),
  updateContact: PropTypes.func,
  removeContact: PropTypes.func,
};

ContactPage.defaultProps = {
  contacts: [],
  match: null,
  updateContact: () => {},
  removeContact: () => {},
};

export default ContactPage;
