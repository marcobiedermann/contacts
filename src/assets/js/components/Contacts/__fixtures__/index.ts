import { ContactsProps } from '..';
import contact from '../../ContactPreview/__fixtures__';

const props: ContactsProps = {
  contacts: [contact],
};

export default props;
