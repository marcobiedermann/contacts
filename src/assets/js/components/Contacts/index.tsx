import React, { FC } from 'react';
import ContactPreview, { ContactPreviewProps } from '../ContactPreview';
import styles from './style.module.css';

export interface ContactsProps {
  contacts: ContactPreviewProps[];
}

const Contacts: FC<ContactsProps> = (props) => {
  const { contacts } = props;

  return (
    <div>
      <ul className={styles.contacts__list}>
        {contacts.map((contact) => (
          <li className={styles.contacts__item} key={contact.id}>
            <ContactPreview {...contact} />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Contacts;
