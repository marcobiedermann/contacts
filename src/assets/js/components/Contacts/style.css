.contacts__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.contacts__item + .contacts__item {
  border-top: 1px solid #b6def5;
}
