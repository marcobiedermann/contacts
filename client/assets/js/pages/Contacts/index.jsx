import React from 'react';
import { Route, Switch } from 'react-router-dom';
import ContactPage from './Contact';
import Contacts from '../../components/Contacts';
import contacts from '../../data/contacts';

const ContactsPage = () => (
  <div>
    <Switch>
      <Route
        path="/contacts/:id"
        component={ContactPage}
      />
      <Route
        path="/contacts"
        component={() => (
          <Contacts contacts={contacts} />
        )}
      />
    </Switch>
  </div>
);

export default ContactsPage;
