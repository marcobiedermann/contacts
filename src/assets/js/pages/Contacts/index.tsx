import React, { lazy } from 'react';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import { Route, Switch } from 'react-router-dom';
import { useFirestore, useFirestoreCollectionData } from 'reactfire';
import Contacts from '../../components/Contacts';
import Loader from '../../components/Loader';
import styles from './style.module.css';

interface Address {
  type: string;
  value: {
    city: string;
    country: string;
    street: string;
    zip: string;
  };
}

interface Email {
  type: string;
  value: string;
}

interface Name {
  firstName: string;
  lastName: string;
}

interface Phone {
  type: string;
  value: string;
}

interface Document {
  addresses: Address[];
  emails: Email[];
  id: string;
  name: Name;
  phones: Phone[];
}

function ContactsPage(): JSX.Element {
  const firestore = useFirestore();
  const ref = firestore.collection('contacts');
  const { data, status } = useFirestoreCollectionData<Document>(ref, {
    idField: 'id',
  });
  const { t } = useTranslation();

  if (status === 'loading') {
    return <Loader />;
  }

  return (
    <div className={styles['contacts-page']}>
      <Helmet>
        <title>{t('pages:contacts')}</title>
      </Helmet>
      <Contacts contacts={data} />
      <Switch>
        <Route path="/contacts/:contactId/edit" component={lazy(() => import('../ContactEdit'))} />
        <Route path="/contacts/new" component={lazy(() => import('../ContactNew'))} />
        <Route path="/contacts/:contactId" component={lazy(() => import('../Contact'))} />
      </Switch>
    </div>
  );
}

export default ContactsPage;
