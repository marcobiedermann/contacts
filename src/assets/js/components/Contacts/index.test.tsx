import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Contacts, { ContactsProps } from '.';

describe('contacts component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const props: ContactsProps = {
      contacts: [
        {
          id: '1',
          name: {
            firstName: 'John',
            lastName: 'Doe',
          },
        },
      ],
    };

    const { asFragment } = render(
      <MemoryRouter>
        <Contacts {...props} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
