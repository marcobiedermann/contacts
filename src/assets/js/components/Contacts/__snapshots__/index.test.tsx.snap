// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contacts component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="contacts__list"
    >
      <li
        class="contacts__item"
      >
        <div>
          <h2
            class="contact__name"
          >
            <a
              class="contact__link"
              href="/contacts/1"
            >
              John Doe
            </a>
          </h2>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
